sdk/storage/Azure.Storage.Common/src/Shared/UserDelegationKeyProperties.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using System.Net;
+using System.Text;
+
+namespace Azure.Storage.Sas
+{
+    /// <summary>
+    /// Encapsulates the shared properties used by both
+    /// BlobSasQueryParameters and DataLakeSasQueryParameters.
+    /// </summary>
+    internal struct UserDelegationKeyProperties
+    {
+        // skoid
+        internal string _objectId;
+
+        // sktid
+        internal string _tenantId;
+
+        // skt
+        internal DateTimeOffset _startsOn;
+
+        // ske
+        internal DateTimeOffset _expiresOn;
+
+        // sks
+        internal string _service;
+
+        // skv
+        internal string _version;
+
+        /// <summary>
+        /// Builds up the UserDelegationKey portion of the SAS query parameter string.
+        /// </summary>
+        public void BuildParameterString(StringBuilder stringBuilder)
```

Comment:
Should we call this `AppendProperties` since you're not building a new string but adding them to an existing builder?

Reply:
I liked having it match the method name of SasQueryParameters.BuildParameterString. It's true that the method in UserDelegationKeyProperties is not meant to be stand-alone where as the one in SasQueryParameters can be stand-alone.  So I guess I would agree with your suggestion.